src/main/java/com/project/videotecha/service/impl/MovieServiceImpl.java

```diff
@@ -26,9 +28,16 @@ public Movie create(Movie movie) {
     @Override
     @Transactional
     public Movie update(Movie movie) {
-        if (movieRepository.findByIdAndDeletedFalse(movie.getId()).isEmpty()) {
-            throw new EntityNotFoundException(MOVIE_NOT_FOUND_MESSAGE + movie.getId());
+        Movie oldMovie = movieRepository.findByIdAndDeletedFalse(movie.getId())
+                .orElseThrow(() -> new EntityNotFoundException(MOVIE_NOT_FOUND_MESSAGE + movie.getId()));
+        List<Projection> projections = oldMovie.getProjections()
+                .stream()
+                .filter(p -> !p.getDeleted() && !hasPassed(p))
+                .toList();
```

Comment:
Instead of filtering and collecting the results into a list, and then checking if the list is empty, try using `anyMatch(...)` method from Stream API